pkg/agent/inteface.go

```diff
@@ -35,6 +35,20 @@ type AgentAddon interface {
 	GetAgentAddonOptions() AgentAddonOptions
 }
 
+type HostedAgentAddon interface {
+	AgentAddon
+	// ManagementManifests returns a list of manifest resources to be deployed on the management cluster for this addon.
+	// The resources in this list are required to explicitly clarify their TypeMta (i.e. apiVersion, kind)
+	// otherwise the addon deploying will fail constantly. A recommended set of addon component returned
+	// here will be:
+	//   - the hosting namespace of the addon-agents.
+	//   - a deployment of the addon agents.
+	//   - the configurations (e.g. configmaps) mounted by the deployment.
+	// NB for dispatching namespaced resources, it's recommended to include the namespace in the list.
+	ManagementManifests(cluster *clusterv1.ManagedCluster, addon *addonapiv1alpha1.ManagedClusterAddOn) (
```

Comment:
I still feel we should have another interface. What is the benefit of doing this?